docs/modules/ROOT/pages/configuration/global-filters.adoc

```diff
@@ -0,0 +1,48 @@
+= Configuring Global Filters on KubeArchive
+
+KubeArchive allows cluster owners to configure a set of filters that are applied
+to any KubeArchive-enabled namespace. To configure them follow the next steps:
+
+1. Create a `KubeArchiveConfig` in the `kubearchive` namespace. The rules
+in this configuration apply to all KubeArchive-enabled namespaces. For example:
++
+[source,yaml]
+----
+---
+apiVersion: kubearchive.kubearchive.org/v1alpha1
+kind: KubeArchiveConfig
+metadata:
+  name: kubearchive
+  namespace: kubearchive
+spec:
+  resources:
+    - selector:
+        apiVersion: v1
+        kind: Pod
+      archiveOnDelete: "true"
+----
+
+1. Create a `KubeArchiveConfig` on the namespace you want the global filters to apply. This
+configuration could be empty or add its own filters as well. Both global and per-namespace
+filters are combined. For example:
++
+[source,yaml]
+----
+---
+apiVersion: kubearchive.kubearchive.org/v1alpha1
+kind: KubeArchiveConfig
+metadata:
+  name: kubearchive
+  namespace: my-team
+spec:
+  resources:
+    - selector:
+        apiVersion: batch/v1
+        kind: Job
+      archiveWhen: has(status.startTime)
+      deleteWhen: has(status.completionTime)
+----
++
+In the namespace `my-team` KubeArchive archives jobs as long as they have `status.startTime`
+and deletes them when they have `status.completionTime`. KubeArchive also archives pods when
+they are about to be deleted, as configured in the global filters.
```

Comment:
As a user I have some questions after reading this example:

* With this example I get that KubeArchiveConfig is "merged" for any namespace with the global one in the `kubearchive` namespace, but does the global KubeArchiveConfig apply on `my-team` namespace if I don't have any `KubeArchiveConfig` configured?

* Also, what if I want `my-team` namespace not to have their pods archived? How should I configure the KubeArchiveConfig in `kubearchive` namespace for that?

I believe that the second question could be answered with a `KubeArchiveConfig` like the following:

```
---
apiVersion: kubearchive.kubearchive.org/v1alpha1
kind: KubeArchiveConfig
metadata:
  name: kubearchive
  namespace: kubearchive
spec:
  resources:
    - selector:
        apiVersion: v1
        kind: Pod
        selector:
          matchExpressions:
            - key: prevent-archive
              operator: DoesNotExist
      archiveOnDelete: "true"
```

And make sure that the pods of the `my-team` namespace have that label applied

Reply:
> With this example I get that KubeArchiveConfig is "merged" for any namespace with the global one in the kubearchive namespace, but does the global KubeArchiveConfig apply on my-team namespace if I don't have any KubeArchiveConfig configured?

The docs say:

>  Create a `KubeArchiveConfig` on the namespace you want the global filters to apply. This
configuration could be empty or add its own filters as well. Both global and per-namespace
filters are combined. For example:

I think that solves the question.

> Also, what if I want my-team namespace not to have their pods archived? How should I configure the KubeArchiveConfig in kubearchive namespace for that?

With this I can be more clear. There is nothing you can do to avoid Pods being archived in this scenario other than removing the filter from the global filters.

Reply:
> I think that solves the question.

Yes, my bad, I overlooked that paragraph.

> With this I can be more clear. There is nothing you can do to avoid Pods being archived in this scenario other than removing the filter from the global filters.

I think it's good to be explicit about that but why not showing also the workaround of the labels?
I have also checked our current [KubeArchiveConfig CRD section](https://kubearchive.github.io/kubearchive/main/configuration/kubearchiveconfig.html) and there is nothing related to the label filtering, any reason not to show an example of the usage?

Reply:
> if the cluster owners are willing to let cluster users avoid their filters.

I don't understand this statement. This is *not* possible.

Global filters are always or'd with the users filters, so they are always in effect. The user cannot control this at all. Essentially, all they can do is add additional filtering.

Reply:
Imagine that the owner of the cluster has configured that a resource of a specific kind, say, `Pods` need to be always deleted and archived when finished.

Let's say me and my team are running some tests on our namespace of a controller that manages the pods. 
I want the pods to be removed by the controller and I don't need the data of those pods because that's not the point of the test.

I may want to talk to the cluster admin and say "hey, please give me a way to label or categorize these pods I am creating so that they aren't managed by kubearchive when they finish". 
If that's the case the admin may want to modify the global filter to provide a label that means "do not archive/delete" this `Pod`.

My point here was to provide an example of this type of global configuration filter for a case like this.

Of course, global filters are added to the users filters, but if the cluster admin *is willing* to provide a label for preventing the archive/deletion of the resource, that could be done.

Reply:
I don't see how this works. Are you saying this is some functionality that `ApiServerSource` provides that would allow exclusion of pods with a particular label?  We have changed to a global (single) `ApiServerSource` instance, and I don't think there's a way there to provide this functionality per-namespace.